src/config.rs

```diff
@@ -92,30 +95,28 @@ impl Config {
     pub fn save_to_path(&self, path: &Path) {
         Self::create_dir(path);
 
-        let file = OpenOptions::new()
-            .write(true)
-            .create(true)
-            .truncate(true)
-            .open(path)
-            .unwrap();
-
-        let writer = io::BufWriter::new(file);
-        serde_json::to_writer(writer, self).unwrap();
+        fs::write(path, toml::to_string(self).unwrap()).unwrap();
     }
 
     pub fn set_watch(&mut self, path: String, cfg: WatchConfig) {
-        if self.repos.contains_key(&path) {
-            println!("{} is already being watched", path)
+        let abs_path = fs::canonicalize(path).expect("The provided path is not a directory");
+        let abs_path = abs_path.to_str().unwrap();
+
+        if self.repos.contains_key(abs_path) {
+            println!("{} is already being watched", abs_path)
         } else {
-            self.repos.insert(path.clone(), Rc::new(cfg));
-            println!("Started watching {}", path)
+            self.repos.insert(abs_path.to_string(), Rc::new(cfg));
+            println!("Started watching {}", abs_path)
         }
     }
 
     pub fn set_unwatch(&mut self, path: String) {
-        match self.repos.remove(&path) {
-            Some(_) => println!("Stopped watching {}", path),
-            None => println!("{} is not being watched", path),
+        let abs_path = fs::canonicalize(path).expect("The provided path is not a directory");
+        let abs_path = abs_path.to_str().unwrap().to_string();
```

Comment:
You sure you need `to_string()` here? Would be a shame to copy unnecessarily, since you're taking a `&` down below anyway :P

Reply:
You're right, that can go actually